### DISCUSSION TITLE
bpf_get_current_task compile error

### DISCUSSION THREAD

**Discussion by liuchenailq** [cilium/ebpf#903] [Q&A] [answered]
I want to get the pid and ppid of the process. My approach is as follows:
```
struct task_struct *task;
struct task_struct *parent;
task = (struct task_struct *)bpf_get_current_task();
bpf_probe_read(&event.ppid, sizeof(event.ppid), &task->real_parent->tgid);
```

But when compiling, I get the following error:
```
error: incomplete definition of type 'struct task_struct'
bpf_probe_read(&event.ppid, sizeof(event.ppid), &task->real_parent->tgid);
~~~~^
../headers/bpf_helper_defs.h:31:8: note: forward declaration of 'struct task_struct'
struct task_struct;
```

I added <linux/sched.h> or reported this error

**Comment by lmb** [comment, marked as answer]
The problem is that your C headers do not contain the full type definition for `struct task_struct`. Instead of including linux headers directly I suggest using [`vmlinux.h`](https://www.grant.pizza/blog/vmlinux-header/).
